validator/client/validator_metrics.go

```diff
@@ -64,10 +65,12 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 			prevBalance := float64(v.prevBalance) / float64(params.BeaconConfig().GweiPerEth)
 			percentNet := (newBalance - prevBalance) / prevBalance
 			log.WithField("prevEthBalance", prevBalance).Infof("%v Previous validator balance", tpk)
-			log.WithFields(logrus.Fields{
-				"eth":           fmt.Sprintf("%f", newBalance-prevBalance),
-				"percentChange": fmt.Sprintf("%.2f%%", percentNet*100),
-			}).Infof("%v Net gains/losses in eth", tpk)
+			if !v.ctxCli.GlobalBool(types.DisablePenaltyRewardLogFlag.Name) {
```

Comment:
this is not the pattern we use, we dont pass cli.Context into the validator struct. You add in a new boolean var into the struct called `LogValidatorBalances` and then you set that when initializing the validator struct. Using the `cli.Context` for the flag we set the value of `LogValidatorBalances` when initialzing the validator struct, and then use that var to determine whether to log balances